fuzzing/fuzz_XmlWriter.cpp

```diff
@@ -0,0 +1,33 @@
+//License: Boost 1.0
+//By Paul Dreik 2020
+
+#include "internal/catch_xmlwriter.h"
+
+
+#include <iostream>
+
+// from https://stackoverflow.com/a/8244052
+class NulStreambuf : public std::streambuf {
+  char dummyBuffer[64];
+
+protected:
+  virtual int overflow(int c) override final{
+    setp(dummyBuffer, dummyBuffer + sizeof(dummyBuffer));
+    return (c == traits_type::eof()) ? '\0' : c;
+  }
+};
+
+class NulOStream final : private NulStreambuf, public std::ostream {
+public:
+  NulOStream() : std::ostream(this) {}
+  NulStreambuf *rdbuf() { return this; }
+};
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+
+    std::string buf(Data,Data+Size);
+    NulOStream nul;
+    Catch::XmlEncode encode(buf);
+    encode.encodeTo(nul);
```

Comment:
I am reasonably sure that this needs an XML validator oracle to catch anything.

Reply:
To know that the xml it outputs is correct? Yes. But I just wanted to exercise the code a bit, I browsed through the cpp files and that was the only one that looked scary.